### PR TITLE
Feature/initial setup

### PR DESCRIPTION
**Pull request by widmogrod**
Address issue https://github.com/functional-php/fantasy-land/issues/1
- Move code from https://github.com/widmogrod/php-functional 
- Change namespaces

Also adds integration with travis https://github.com/functional-php/fantasy-land/issues/5
- code style checks

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+language: php
+
+php:
+ - 7.1
+ - 7.2
```
Comment (krtek4) [inline .travis.yml]: We might want to support PHP 5.6 ? (EOL 01.01.2019)

Or are there already some features in the code that are incompatible ?

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  "type": "library",
+  "license": "BSD-3-Clause",
+  "authors": [
+    {
+      "name": "Gabriel Habryn",
+      "email": "widmogrod+github@gmail.com"
+    }
+  ],
+  "require": {
+    "php": ">=7.1",
```
Comment (krtek4) [inline composer.json]: See above about possible PHP  version change

**Comment by krtek4** [comment]
Seems all good to me :)

It seems that the `declare(strict_types=1);` is incompatible with PHP5.6 which is kind of a shame. It still representes roughly 30% of the user base.

I think we can keep it at PHP7. let me know what you think :)

File: .travis.yml (hunk 1 above)
Comment (widmogrod) [reply .travis.yml]: Except the `return type` nothing new yet.

**Comment by widmogrod** [comment]
@krtek4 very good, and hard question.

Doing it now is also an option, there is no performance gain by using `declare(strict_types=1);`, so it can be easily removed. Only harm would be regarding `Functor` which depends on return type `: self`

We could make assumption that IF there is an need from the community to support older version of PHP we should do it. 

Doing it now it's little bit like "premature optimisation" build on assumption that we will loose user because we don't support PHP 5.6.

**Comment by krtek4** [comment]
Ok, seems right :)

**Pull request merged**